draft-ietf-httpbis-cache-latest.xml

```diff
@@ -1080,13 +1080,38 @@
    response, with the following exceptions:
 </t>
 <ul>
-   <li>The exceptions to header field storage in <xref
-   target="storing.fields"/> also apply to header field updates.</li>
-   <li>Caches &MUST-NOT; update the following header fields:
-   <x:ref>Content-Encoding</x:ref>, <x:ref>Content-Length</x:ref>,
-   Content-MD5 (<xref target="RFC2616" x:fmt="of" x:sec="14.15"/>),
-   <x:ref>Content-Range</x:ref>, <x:ref>ETag</x:ref>.</li>
+   <li>Header fields excepted from storage in <xref target="storing.fields"/>,</li>
+   <li>Header fields that the cache's stored representation of the response depends upon, as described below,</li>
+   <li>Header fields that are automatically processed and removed by the recipient, as described below, and</li>
+   <li>The Content-Length header field.</li>
 </ul>
+<t>
+   In some cases, caches (especially in user agents) store processed
+   representations of the received response, rather than the response itself,
+   and updating header fields that affect that processing can result in
+   inconsistent behaviour and security issues. Caches in this situation &MAY;
+   omit these header fields from updating stored representations on an
+   exceptional basis, but &SHOULD; limit such omission to those fields
+   necessary to assure integrity of the stored representation.
+</t>
+<t>
+   For example, a browser cache might store a response after removing
+   content-codings; updating that response with a different Content-Encoding
+   header field would be problematic. Likewise, a browser cache might store a
+   post-parse tree representation of HTML, rather than the bytes received in
+   the response; updating the Content-Type header field would not be workable
+   in this case.
+</t>
```

Comment:
I feel this is a bit confusing, since the browser is usually not acting as a cache when it stores data transformed from a response. Why is this part relevant to HTTP?

A better way to think of it is the original response exists with its data portion transformed and stored by reference. The data is no longer with the response; HTTP does not care because this is a user agent. However, the metadata (and anything else the browser chose to store about its original request/response) is stored like a cached response and can be validated/updated like a cached response. A browser can use that stored metadata and request context when choosing to update (or not) its stored response, whether or not the received representation was transformed, provided that validation uses the original (pre-transformed) response fields.

Reply:
Hence, when a successful validation response (e.g., 304) is received with new field information, the browser needs to interpret that information with respect to the original request and its original metadata.

Reply:
IOW, all cache validations are based on field metadata; no cache operates by sending the data stream back to the server to see if it is still valid. Hence, how the response data is processed or stored after the response is received is not relevant to HTTP so long as the recipient keeps the original metadata for later validation requests.

Reply:
```suggestion
<t>
   For example, a browser might store a response's body after removing
   content-codings, thereby making its metadata inaccurate. 
   Updating that stored metadata with a different Content-Encoding
   header field would be problematic. Likewise, a browser might store a
   post-parse tree representation of HTML, rather than the body received in
   the response; updating the Content-Type header field would not be workable
   in this case, because any assumptions about the format made in parsing would
   now be invalid.
</t>
```

@royfielding PTAL